ksqldb-execution/src/main/java/io/confluent/ksql/schema/query/QuerySchemas.java

```diff
@@ -118,6 +132,60 @@ public synchronized void trackSerdeOp(
     return Collections.unmodifiableMap(loggerToSchemas);
   }
 
+  private SchemaInfo getKeySchemaInfo(final Map<Boolean, Set<String>> kvLoggerNames,
+                                      final String topicName) {
+    final Set<String> topicNames = kvLoggerNames.getOrDefault(KEY_LOGGER_TYPE, ImmutableSet.of());
+
+    if (topicNames.size() != 1) {
+      final String result = topicNames.size() == 0 ? "Zero" : "Multiple";
+
+      throw new IllegalStateException(result + " key logger names registered for topic."
+          + System.lineSeparator()
+          + "topic: " + topicName
+          + "loggers: " + topicNames
+      );
+    }
+
+    return loggerToSchemas.get(Iterables.getOnlyElement(topicNames));
+  }
+
+  private SchemaInfo getValueSchemaInfo(final Map<Boolean, Set<String>> kvLoggerNames,
+                                        final String topicName) {
+    final Set<String> topicNames = kvLoggerNames.getOrDefault(VALUE_LOGGER_TYPE, ImmutableSet.of());
+
+    // NO_VALUE_SCHEMA_FOUND is returned if no value topic has been detected yet. This happens
+    // when a null value (delete) was processed in the topic as the first record. The internal
+    // serde from the value topic does not call the TrackSerde, which can't track the format type
+    // for the topic.
+    final SchemaInfo valueInfo = loggerToSchemas.getOrDefault(
+        Iterables.getFirst(topicNames, EMPTY_TOPIC_NAME),
+        NO_VALUE_SCHEMA_FOUND);
+
+    if (topicNames.size() > 1) {
+      final Optional<ValueFormat> firstValueFormat = valueInfo.valueFormat();
+
+      // Multiple value topics names may be detected. This happens with left/outer joins. The left
+      // and right streams share a state store which share one changelog topic with different
+      // value formats. The track code, though, is only used for QTT which always executes joins
+      // with same value formats. We only make sure all formats are the same so we return one
```

Comment:
> is only used for QTT which always executes joins with same value formats

This seems to be the way the test are effectively written but not a fundamental limitation. At it seems QTT should be able to test all valid queries and joining two streams with different value formats seems to be a valid query that we support atm, and should keep supporting?

Reply:
+1 seems like something we should support and test. if there's a bug here then there might be real bugs in the code somewhere. out of scope for this PR but can you create a ticket tracking it?

Reply:
+1 to the confusion above. Is it true that this code only affects the intermediate topic value formats written into the historic plan spec files, and that the lack of support for differing value formats in stream-stream joins introduced here is only limited to what gets written into the spec, rather than our ability to run such tests and verify inputs and outputs (assuming we were to remove the error thrown on line 177 below)?